docs/spec.md

```diff
@@ -4935,20 +4935,16 @@ def sign(x):
   if is_integer(x):
     if compare(x, 0, LT, SIGNED): return -1
     if compare(x, 0, EQ, SIGNED): return 0
-    if compare(x, 0, GT, SIGNED): return 1
+    return 1
   elif is_float(x):
-    if x is NaN:
-      return NaN
-    else:
-      if compare(x, 0.0, LT, FLOAT): return -1.0
-      if compare(x, -0.0, EQ, FLOAT): return -0.0
-      if compare(x, +0.0, EQ, FLOAT): return +0.0
-      if compare(x, 0.0, GT, FLOAT): return 1.0
+    if x is NaN: return NaN
+    if compare(x, -0.0, EQ, FLOAT): return -0.0
+    if compare(x, +0.0, EQ, FLOAT): return +0.0
+    if compare(x, 0.0, LT, FLOAT): return -1.0
+    return 1.0
   elif is_complex(x):
-    if x.real is NaN or x.imag is NaN:
-      return NaN
-    else:
-      return divide(x, abs(x))
+    if x.real is NaN or x.imag is NaN: return (NaN, NaN)
+    return divide(x, abs(x))
```

Comment:
abs(x) returns a float, so how about explicitly writing:
```
return divide(x, (abs(x).real, abs(x).imag))
```

Reply:
"abs(x) returns a float". Good point, we've also discussed this in #1382. You probably meant `divide(x, (abs(x), 0))`, right? Anyway, I used convert instead to hopefully make this more explicit.

"We should also have a separate case for handling zeros (regardless of sign should return positive zero)". Is this behavior specced somewhere? Here are the data points that seem relevant:
  * [elemental_ir_emitter.py](https://github.com/tensorflow/tensorflow/blob/f6ca332b93cc5584dd09ca991865a6a7091f3eff/tensorflow/compiler/xla/service/elemental_ir_emitter.cc#L1054-L1064) does what's proposed in the comment above.
  * [lax.py](https://github.com/google/jax/blob/d9383fc80dc739443fd10365c85c7e781e370b01/jax/_src/lax/lax.py#L234-L235) only says `x / |x|`.
  * Surprisingly, [numpy.sign](https://numpy.org/doc/stable/reference/generated/numpy.sign.html) does something quite different: `sign(x.real) + 0j if x.real != 0 else sign(x.imag) + 0j`, so we shouldn't probably index too much on that.

Reply:
Ah yes, I meant `(abs(x), 0)`. Handling zeros is a specific behavior in `elemental_ir_emitter.py`. And though the jax doc you linked seem to suggest otherwise, as mentioned in https://github.com/openxla/stablehlo/pull/1382#discussion_r1165960432, lowering to hlo in [lax.py](https://github.com/google/jax/blob/d9383fc80dc739443fd10365c85c7e781e370b01/jax/_src/lax/lax.py#L1714-L1724) where the `sign_p` comes from the [line](https://github.com/google/jax/blob/d9383fc80dc739443fd10365c85c7e781e370b01/jax/_src/lax/lax.py#L237) under the doc also follows this behavior.

Reply:
Upon further reflection, I am convinced now that it is a good idea to specialcase `(0.0, 0.0)` in the specification, consistently with how `0.0` is specialcased as well. I've updated the PR accordingly.